addons/ShaderFunction-Extras/Utility/utility.gdshaderinc

```diff
@@ -28,6 +28,47 @@ float proximity_fade_gles(sampler2D depth_tex, vec2 screen_uv, mat4 inv_proj_mat
 	return clamp(1.0 - smoothstep(depth_world_pos.z + fade_dist, depth_world_pos.z, vertex.z), 0.0, 1.0);
 }
 
+vec3 world_position_from_depth(sampler2D depth_tex, vec2 screen_uv, mat4 inv_proj_mat, mat4 inv_view_matrix) {
+	float log_depth = textureLod(depth_tex, screen_uv, 0.0).x;
+	vec3 depth_ndc = vec3(SCREEN_UV * 2.0 - 1.0, log_depth);
```

Comment:
this probably also needs a seperate method to calculate the ndc for Vulkan and GLES3. I may need to patch my PR in Godot core as well